pandas/tests/series/test_dtypes.py

```diff
@@ -153,6 +153,38 @@ def test_astype_dict(self):
         self.assertRaises(KeyError, s.astype, {'abc': str, 'def': str})
         self.assertRaises(KeyError, s.astype, {0: str})
 
+    def test_astype_generic_timestamp(self):
+        # see gh-15524
+        data = [1]
+
+        s = Series(data)
+        dtype = np.datetime64
+        result = s.astype(dtype)
+        expected = Series(data, dtype=dtype)
+        assert_series_equal(result, expected)
+
+        s = Series(data)
+        dtype = np.timedelta64
+        result = s.astype(dtype)
+        expected = Series(data, dtype=dtype)
+        assert_series_equal(result, expected)
+
+    def test_astype_empty_constructor_equality(self):
+        # see gh-15524
+
+        for dtype in np.typecodes['All']:
```

Comment:
What's wrong with writing our own decorator for this?  As I said before, we did exactly this for capturing `stdout` and `stderr`.

Also, having had to "rewrite" tests in `pytest` idiom instead of `unittest` idiom, I've come to believe that the testing interface should be framework-agnostic (i.e. we just call `tm.<something>` all the time and don't worry about what the implementation or framework is underneath).